### ISSUE TITLE
Misfit Model May Produce NaN confidence interval

### ISSUE THREAD

**Issue by davidkowalk** [davidkowalk/fitting_toolkit#22]
Fitting a wrong model may produce an infinite covariance matrix, which fills the confidence interval with NaN values

# Steps to Reproduce

```python
import numpy as np
from fitting_toolkit import curve_fit

x= np.linspace(0, 2, 50)
y = x**x

def f(x, a, b):
     return a*np.exp(x + b)

p,c,l,u = curve_fit(f, x, y)

print(c)
print(f(x, *p))
```

# Expected Behaviour

An Error or a Warning should be raised. Furthermore the most likely model is not found.

# Observed Behaviour

```python
>>> print(c)
[[inf inf]
 [inf inf]]
>>> print(f(x, *p))
[6.70692142e-42 6.98633688e-42 7.27739300e-42 7.58057475e-42
 7.89638727e-42 8.22535678e-42 8.56803142e-42 8.92498214e-42
 9.29680370e-42 9.68411564e-42 1.00875633e-41 1.05078189e-41
 1.09455826e-41 1.14015839e-41 1.18765826e-41 1.23713701e-41
 1.28867708e-41 1.34236435e-41 1.39828827e-41 1.45654203e-41
 1.51722268e-41 1.58043133e-41 1.64627331e-41 1.71485831e-41
 1.78630061e-41 1.86071926e-41 1.93823824e-41 2.01898673e-41
 2.10309925e-41 2.19071597e-41 2.28198287e-41 2.37705201e-41
 2.47608181e-41 2.57923727e-41 2.68669026e-41 2.79861983e-41
 2.91521246e-41 3.03666244e-41 3.16317211e-41 3.29495227e-41
 3.43222250e-41 3.57521150e-41 3.72415754e-41 3.87930878e-41
 4.04092374e-41 4.20927171e-41 4.38463318e-41 4.56730034e-41
 4.75757755e-41 4.95578186e-41]
```

**Comment by davidkowalk** [comment]
2f427eb08b76eb59bc52e32e864a2d4a6b34e425 adds runtime warning, tests to be added.

**Comment by davidkowalk** [comment]
Closed By 2c03c2d660a25119f5f73cd524e68412a5cc5079
